IPython/core/interactiveshell.py

```diff
@@ -2711,14 +2723,14 @@ def get_cells():
         with prepended_to_syspath(dname):
             try:
                 for cell in get_cells():
-                    # self.run_cell currently captures all exceptions
-                    # raised in user code.  It would be nice if there were
-                    # versions of run_cell that did raise, so
-                    # we could catch the errors.
                     result = self.run_cell(cell, silent=True, shell_futures=shell_futures)
-                    if not result.success:
+                    if not result.success and raise_exceptions:
+                        result.raise_error()
+                    elif not result.success:
```

Comment:
I think this is now checking the exceptions up to four times. We can simplify it by doing:

```python
if raise_exceptions:
    result.raise_error()
elif not result.success:
    break
```